assignments/callbacks.js

```diff
@@ -24,29 +24,72 @@ const items = ['Pencil', 'Notebook', 'yo-yo', 'Gum'];
 
 function getLength(arr, cb) {
   // getLength passes the length of the array into the callback.
+  return cb(arr.length);
 }
 
 function last(arr, cb) {
   // last passes the last item of the array into the callback.
+   return cb(arr[arr.length -1]);
 }
 
 function sumNums(x, y, cb) {
   // sumNums adds two numbers (x, y) and passes the result to the callback.
+  let sum = x + y;
+   return cb(sum);
 }
 
 function multiplyNums(x, y, cb) {
   // multiplyNums multiplies two numbers and passes the result to the callback.
+  let product =  x*y;
+  return cb(product);
 }
 
 function contains(item, list, cb) {
   // contains checks if an item is present inside of the given array/list.
   // Pass true to the callback if it is, otherwise pass false.
+ 
+  // for (let i = 0; i < list.length; i++) {
+  //   const element = list[i];
+  //   if(item === element){
+  //     cb(true)
+  //   } else {
+  //     cb(false)
+  //   }
+  // }
+
+  list.forEach(element => {item === element ? cb(true) : cb(false)});
 }
 
+let caller = call => { call };
+
+console.log(getLength(items, caller));
+console.log(last(items, caller));
+console.log(sumNums(12,12,caller));
+console.log(multiplyNums(12,12,caller));
+console.log(contains('Notebook', items, caller))
+
+
 /* STRETCH PROBLEM */
 
+const names = ["Mike","Matt","Nancy","Adam","Jenny","Nancy","Carl","Adam"];
+
 function removeDuplicates(array, cb) {
   // removeDuplicates removes all duplicate values from the given array.
   // Pass the duplicate free array to the callback function.
   // Do not mutate the original array.
+  let newArr = [];
+ 
+  // for (let i = 0; i < array.length; i++) {
+  //   for(let j = array.length; j >= 0; j--){
+  //     if(array[i] != array[j]){
+  //     newArr.push(array[i])
+  //     }
+  //   }  
+  // }
+
+  newArr = names.filter(()=>{});
```

Comment:
make sure to come back and finish this one, I actually did a similar problem for the whiteboarding demo. don't overcomplicate it, just look at how you might be able to use `.includes()` with this problem